Library/Homebrew/dev-cmd/generate-cask-api.rb

```diff
@@ -3,6 +3,7 @@
 
 require "cli/parser"
 require "cask/cask"
+require "formula"
```

Comment:
Where does this come from? In theory casks and formulae should be separate.

Reply:
It comes in via requiring `livecheck` transitively: https://github.com/Homebrew/brew/blob/4.0.29/Library/Homebrew/livecheck.rb#L22

We only see this with sorbet enabled, as otherwise the block is not evaluated. The `Formula` reference there is needed bc otherwise:
```
$ brew typecheck
formula.rb:2822: Expected T.any(Cask::Cask, Resource) but found T.class_of(Formula) for argument package_or_resource https://srb.help/7002
    2822 |        @livecheck = Livecheck.new(self)
                                             ^^^^
  Expected T.any(Cask::Cask, Resource) for argument package_or_resource of method Livecheck#initialize:
    livecheck.rb:22:
    22 |  sig { params(package_or_resource: T.any(Cask::Cask, Resource)).void }
```

(I'm not positive this is answering your question though)

Reply:
Oh right, the type signatures.

This is probably good enough for now (assuming `brew livecheck` and `brew audit --cask` on a cask with a livecheck block works), but I guess in the longer term we should have a baseclass for any sort of package/resource. Might be the next big refactor work I can look into.